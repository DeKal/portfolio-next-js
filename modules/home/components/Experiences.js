import React from 'react'
import { EXPERIENCES } from '~/modules/home/consts/pages'
import { getCompanies } from '~/modules/home/consts/experiences'
import Header from '~/modules/home/components/common/section/Header'
import SectionContainer from '~/modules/home/components/common/section/Container'
import List from '~/modules/home/components/experiences/List'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Anchor from '~/modules/home/components/common/Anchor'
import { Translate } from 'react-localize-redux'

const companyRows = getCompanies()

const Experiences = () => (
  <SectionContainer>
    <Anchor id={EXPERIENCES} top={-40} />
    <Container>
      <Header>
        <h2>
          <Translate id="Exp: Title" />
        </h2>
        <h4>
          <Translate id="Exp: SubTitle" />
        </h4>
      </Header>
      {companyRows.map((companies, index) => (
        <Row key={index}>
          <List companies={companies} />
        </Row>
      ))}
    </Container>
  </SectionContainer>
)

export default Experiences
