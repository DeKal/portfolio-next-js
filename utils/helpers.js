import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
