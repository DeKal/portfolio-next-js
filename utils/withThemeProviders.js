import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
