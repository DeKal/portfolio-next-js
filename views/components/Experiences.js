import React from 'react'
import { EXPERIENCES } from '~/consts/pages'
import { getCompanies } from '~/consts/experiences'
import Header from '~/views/components/common/section/Header'
import SectionContainer from '~/views/components/common/section/Container'
import List from '~/views/components/experiences/List'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Anchor from '~/views/components/common/Anchor'
import { Translate } from 'react-localize-redux'

const companyRows = getCompanies()

const Experiences = () => (
  <SectionContainer>
    <Anchor id={EXPERIENCES} top={-40} />
    <Container>
      <Header>
        <h2>
          <Translate id="Exp: Title" />
        </h2>
        <h4>
          <Translate id="Exp: SubTitle" />
        </h4>
      </Header>
      {companyRows.map((companies, index) => (
        <Row key={index}>
          <List companies={companies} />
        </Row>
      ))}
    </Container>
  </SectionContainer>
)

export default Experiences
