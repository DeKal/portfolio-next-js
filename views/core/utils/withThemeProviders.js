import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
