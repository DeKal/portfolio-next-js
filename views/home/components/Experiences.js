import React from 'react'
import { EXPERIENCES } from '~/views/home/consts/pages'
import { getCompanies } from '~/views/home/consts/experiences'
import Header from '~/views/home/components/common/section/Header'
import SectionContainer from '~/views/home/components/common/section/Container'
import List from '~/views/home/components/experiences/List'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Anchor from '~/views/home/components/common/Anchor'
import { Translate } from 'react-localize-redux'

const companyRows = getCompanies()

const Experiences = () => (
  <SectionContainer>
    <Anchor id={EXPERIENCES} top={-40} />
    <Container>
      <Header>
        <h2>
          <Translate id="Exp: Title" />
        </h2>
        <h4>
          <Translate id="Exp: SubTitle" />
        </h4>
      </Header>
      {companyRows.map((companies, index) => (
        <Row key={index}>
          <List companies={companies} />
        </Row>
      ))}
    </Container>
  </SectionContainer>
)

export default Experiences
