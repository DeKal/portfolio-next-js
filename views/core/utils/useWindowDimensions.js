import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
