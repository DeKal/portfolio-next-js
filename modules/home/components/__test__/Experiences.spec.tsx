/* eslint-disable react/prop-types */
import * as React from 'react'
import { mountWithTheme } from '~/modules/core/utils/withThemeProviders'
import Experiences from '~/modules/home/components/Experiences'

const MockVerticalTimelineElement = () => <div />

jest.mock('react-vertical-timeline-component', () => ({
  VerticalTimeline: ({ children }) => children,
  VerticalTimelineElement: () => MockVerticalTimelineElement
}))

describe('Experiences', () => {
  it('should render without throwing an error', function() {
    const wrap = mountWithTheme(<Experiences />)
    expect(wrap).toMatchSnapshot()
  })
})
