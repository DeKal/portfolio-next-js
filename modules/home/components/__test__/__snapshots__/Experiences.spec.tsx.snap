// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiences should render without throwing an error 1`] = `
<Experiences>
  <Container>
    <Container__Style>
      <section
        className="Container__Style-sc-3utn0w-0 fovYsq"
      >
        <Anchor
          id="Experiences"
          top={-125}
        >
          <span
            className="Anchor-sc-1axmo0z-0 fyFXXM"
            id="Experiences"
          />
        </Anchor>
        <Container
          fluid={false}
        >
          <div
            className="container"
          >
            <Header>
              <Header__Container>
                <header
                  className="Header__Container-hz47ou-0 dxCyVk"
                >
                  <h2>
                    <Translate
                      id="Exp: Title"
                    >
                      <WrappedTranslate
                        context={
                          Object {
                            "activeLanguage": undefined,
                            "addTranslation": [Function],
                            "addTranslationForLanguage": [Function],
                            "defaultLanguage": undefined,
                            "ignoreTranslateChildren": false,
                            "initialize": [Function],
                            "languages": Array [],
                            "renderToStaticMarkup": false,
                            "setActiveLanguage": [Function],
                            "translate": [Function],
                          }
                        }
                        id="Exp: Title"
                      >
                        Missing translationId: Exp: Title for language: \${ languageCode }
                      </WrappedTranslate>
                    </Translate>
                  </h2>
                  <h4>
                    <Translate
                      id="Exp: SubTitle"
                    >
                      <WrappedTranslate
                        context={
                          Object {
                            "activeLanguage": undefined,
                            "addTranslation": [Function],
                            "addTranslationForLanguage": [Function],
                            "defaultLanguage": undefined,
                            "ignoreTranslateChildren": false,
                            "initialize": [Function],
                            "languages": Array [],
                            "renderToStaticMarkup": false,
                            "setActiveLanguage": [Function],
                            "translate": [Function],
                          }
                        }
                        id="Exp: SubTitle"
                      >
                        Missing translationId: Exp: SubTitle for language: \${ languageCode }
                      </WrappedTranslate>
                    </Translate>
                  </h4>
                </header>
              </Header__Container>
            </Header>
            <VerticalTimeline
              animate={false}
              className="vertical-timeline-experience-line"
            >
              <Experience
                company="Zalora"
                icon="/images/companies/zalora.webp"
                key="0"
              >
                <VerticalTimelineElement
                  className="vertical-timeline-element--work"
                  contentArrowStyle={
                    Object {
                      "borderRight": "10px solid  #f3f6f6",
                    }
                  }
                  contentStyle={
                    Object {
                      "background": "#f3f6f6",
                      "borderTop": "4px solid  #FFA500",
                      "paddingLeft": 25,
                    }
                  }
                  date={
                    <Experience__Dates>
                      <Translate
                        id="Exp: Zalora Dates"
                      />
                    </Experience__Dates>
                  }
                  icon={
                    <Experience__Logo
                      alt="Zalora"
                      src="/images/companies/zalora.webp"
                    />
                  }
                  iconStyle={
                    Object {
                      "alignItems": "center",
                      "background": "#f8f8ff ",
                      "boxShadow": "0 0 0 3px #FFA500, inset 0 2px 0 rgba(0,0,0,.08), 0 3px 0 4px rgba(0,0,0,.05)",
                      "color": "black",
                      "display": "flex",
                      "justifyContent": "center",
                      "overflow": "hidden",
                    }
                  }
                />
              </Experience>
              <Experience
                company="Dek"
                icon="/images/companies/dek.webp"
                key="1"
              >
                <VerticalTimelineElement
                  className="vertical-timeline-element--work"
                  contentArrowStyle={
                    Object {
                      "borderRight": "10px solid  #f3f6f6",
                    }
                  }
                  contentStyle={
                    Object {
                      "background": "#f3f6f6",
                      "borderTop": "4px solid  #FFA500",
                      "paddingLeft": 25,
                    }
                  }
                  date={
                    <Experience__Dates>
                      <Translate
                        id="Exp: Dek Dates"
                      />
                    </Experience__Dates>
                  }
                  icon={
                    <Experience__Logo
                      alt="Dek"
                      src="/images/companies/dek.webp"
                    />
                  }
                  iconStyle={
                    Object {
                      "alignItems": "center",
                      "background": "#f8f8ff ",
                      "boxShadow": "0 0 0 3px #FFA500, inset 0 2px 0 rgba(0,0,0,.08), 0 3px 0 4px rgba(0,0,0,.05)",
                      "color": "black",
                      "display": "flex",
                      "justifyContent": "center",
                      "overflow": "hidden",
                    }
                  }
                />
              </Experience>
              <Experience
                company="GCS"
                icon="/images/companies/gcs.webp"
                key="2"
              >
                <VerticalTimelineElement
                  className="vertical-timeline-element--work"
                  contentArrowStyle={
                    Object {
                      "borderRight": "10px solid  #f3f6f6",
                    }
                  }
                  contentStyle={
                    Object {
                      "background": "#f3f6f6",
                      "borderTop": "4px solid  #FFA500",
                      "paddingLeft": 25,
                    }
                  }
                  date={
                    <Experience__Dates>
                      <Translate
                        id="Exp: GCS Dates"
                      />
                    </Experience__Dates>
                  }
                  icon={
                    <Experience__Logo
                      alt="GCS"
                      src="/images/companies/gcs.webp"
                    />
                  }
                  iconStyle={
                    Object {
                      "alignItems": "center",
                      "background": "#f8f8ff ",
                      "boxShadow": "0 0 0 3px #FFA500, inset 0 2px 0 rgba(0,0,0,.08), 0 3px 0 4px rgba(0,0,0,.05)",
                      "color": "black",
                      "display": "flex",
                      "justifyContent": "center",
                      "overflow": "hidden",
                    }
                  }
                />
              </Experience>
              <Experience
                company="GCS-intern"
                icon="/images/companies/gcs.webp"
                key="3"
              >
                <VerticalTimelineElement
                  className="vertical-timeline-element--work"
                  contentArrowStyle={
                    Object {
                      "borderRight": "10px solid  #f3f6f6",
                    }
                  }
                  contentStyle={
                    Object {
                      "background": "#f3f6f6",
                      "borderTop": "4px solid  #FFA500",
                      "paddingLeft": 25,
                    }
                  }
                  date={
                    <Experience__Dates>
                      <Translate
                        id="Exp: GCS-intern Dates"
                      />
                    </Experience__Dates>
                  }
                  icon={
                    <Experience__Logo
                      alt="GCS-intern"
                      src="/images/companies/gcs.webp"
                    />
                  }
                  iconStyle={
                    Object {
                      "alignItems": "center",
                      "background": "#f8f8ff ",
                      "boxShadow": "0 0 0 3px #FFA500, inset 0 2px 0 rgba(0,0,0,.08), 0 3px 0 4px rgba(0,0,0,.05)",
                      "color": "black",
                      "display": "flex",
                      "justifyContent": "center",
                      "overflow": "hidden",
                    }
                  }
                />
              </Experience>
              <StartExp>
                <VerticalTimelineElement
                  icon={<StartLogo />}
                  iconStyle={
                    Object {
                      "boxShadow": "0 0 0 3px rgb(16, 204, 82), inset 0 2px 0 rgb(16, 204, 82), 0 3px 0 4px rgb(16, 204, 82)",
                    }
                  }
                />
              </StartExp>
            </VerticalTimeline>
          </div>
        </Container>
      </section>
    </Container__Style>
  </Container>
</Experiences>
`;
